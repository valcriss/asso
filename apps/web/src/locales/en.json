{
  "app": {
    "name": "Asso",
    "tagline": "Management & Accounting",
    "footer": {
      "copyright": "© {year} Asso. All rights reserved.",
      "links": {
        "legal": "Legal notices",
        "privacy": "Privacy policy",
        "support": "Support"
      }
    },
    "navigation": {
      "title": "Navigation",
      "main": "Main navigation",
      "sidebar": "Sidebar navigation",
      "dashboard": "Dashboard",
      "accounting": "Accounting",
      "members": "Members",
      "projects": "Projects",
      "grants": "Grants",
      "memberPortal": "Member portal",
      "supervision": "Supervision",
      "settings": "Settings"
    }
  },
  "auth": {
    "logout": "Sign out"
  },
  "common": {
    "loading": "Loading…",
    "modal": {
      "close": "Close"
    }
  },
  "actions": {
    "newEntry": "New entry",
    "save": "Save",
    "cancel": "Cancel"
  },
  "settings": {
    "pageTitle": "Preferences",
    "pageDescription": "Manage localization and accounting defaults for your workspace.",
    "sections": {
      "locale": {
        "title": "Interface language",
        "description": "Choose the language used across the application.",
        "selectLabel": "Language"
      },
      "currency": {
        "title": "Accounting currency",
        "description": "Amounts are formatted using the selected currency.",
        "selectLabel": "Currency",
        "preview": "Preview: {amount} on {date}"
      }
    },
    "options": {
      "locales": {
        "fr": "French",
        "en": "English"
      },
      "currencies": {
        "EUR": "Euro (EUR)",
        "USD": "US Dollar (USD)",
        "GBP": "Pound Sterling (GBP)"
      }
    },
    "notifications": {
      "saved": "Preferences updated"
    }
  },
  "layout": {
    "openNavigation": "Open navigation",
    "close": "Close"
  }
}
