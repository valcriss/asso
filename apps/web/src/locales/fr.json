{
  "app": {
    "name": "Asso",
    "tagline": "Gestion & Comptabilité",
    "footer": {
      "copyright": "© {year} Asso. Tous droits réservés.",
      "links": {
        "legal": "Mentions légales",
        "privacy": "Politique de confidentialité",
        "support": "Support"
      }
    },
    "navigation": {
      "title": "Navigation",
      "main": "Navigation principale",
      "sidebar": "Navigation latérale",
      "dashboard": "Tableau de bord",
      "accounting": "Comptabilité",
      "members": "Membres",
      "projects": "Projets",
      "grants": "Subventions",
      "memberPortal": "Portail adhérent",
      "supervision": "Supervision",
      "settings": "Paramètres"
    }
  },
  "auth": {
    "logout": "Se déconnecter"
  },
  "common": {
    "loading": "Chargement…",
    "modal": {
      "close": "Fermer"
    }
  },
  "actions": {
    "newEntry": "Nouvelle écriture",
    "save": "Enregistrer",
    "cancel": "Annuler"
  },
  "settings": {
    "pageTitle": "Préférences",
    "pageDescription": "Configurez la langue et la devise utilisées dans l'application.",
    "sections": {
      "locale": {
        "title": "Langue de l'interface",
        "description": "Choisissez la langue utilisée dans l'interface.",
        "selectLabel": "Langue"
      },
      "currency": {
        "title": "Devise comptable",
        "description": "Les montants sont formatés avec la devise sélectionnée.",
        "selectLabel": "Devise",
        "preview": "Aperçu : {amount} au {date}"
      }
    },
    "options": {
      "locales": {
        "fr": "Français",
        "en": "Anglais"
      },
      "currencies": {
        "EUR": "Euro (EUR)",
        "USD": "Dollar américain (USD)",
        "GBP": "Livre sterling (GBP)"
      }
    },
    "notifications": {
      "saved": "Préférences mises à jour"
    }
  },
  "layout": {
    "openNavigation": "Ouvrir la navigation",
    "close": "Fermer"
  }
}
