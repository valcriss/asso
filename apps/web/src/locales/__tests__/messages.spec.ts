import { describe, expect, it } from 'vitest';

import en from '../en.json';
import fr from '../fr.json';

type Messages = Record<string, unknown>;

type FlatMessages = Record<string, string>;

function flattenMessages(messages: Messages, prefix = ''): FlatMessages {
  return Object.entries(messages).reduce<FlatMessages>((acc, [key, value]) => {
    const nextKey = prefix ? `${prefix}.${key}` : key;
    if (value && typeof value === 'object' && !Array.isArray(value)) {
      Object.assign(acc, flattenMessages(value as Messages, nextKey));
    } else if (typeof value === 'string') {
      acc[nextKey] = value;
    }
    return acc;
  }, {});
}

describe('i18n messages', () => {
  it('share the same structure between locales', () => {
    const enFlat = flattenMessages(en);
    const frFlat = flattenMessages(fr);

    expect(Object.keys(frFlat).sort()).toEqual(Object.keys(enFlat).sort());
  });

  it('matches the snapshot for English translations', () => {
    const enFlat = flattenMessages(en);
    expect(enFlat).toMatchSnapshot();
  });

  it('matches the snapshot for French translations', () => {
    const frFlat = flattenMessages(fr);
    expect(frFlat).toMatchSnapshot();
  });
});
