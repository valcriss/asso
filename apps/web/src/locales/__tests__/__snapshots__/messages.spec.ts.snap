// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`i18n messages > matches the snapshot for English translations 1`] = `
{
  "actions.cancel": "Cancel",
  "actions.newEntry": "New entry",
  "actions.save": "Save",
  "app.footer.copyright": "© {year} Asso. All rights reserved.",
  "app.footer.links.legal": "Legal notices",
  "app.footer.links.privacy": "Privacy policy",
  "app.footer.links.support": "Support",
  "app.name": "Asso",
  "app.navigation.accounting": "Accounting",
  "app.navigation.dashboard": "Dashboard",
  "app.navigation.grants": "Grants",
  "app.navigation.main": "Main navigation",
  "app.navigation.memberPortal": "Member portal",
  "app.navigation.members": "Members",
  "app.navigation.projects": "Projects",
  "app.navigation.settings": "Settings",
  "app.navigation.sidebar": "Sidebar navigation",
  "app.navigation.supervision": "Supervision",
  "app.navigation.title": "Navigation",
  "app.tagline": "Management & Accounting",
  "auth.logout": "Sign out",
  "common.loading": "Loading…",
  "common.modal.close": "Close",
  "layout.close": "Close",
  "layout.openNavigation": "Open navigation",
  "settings.notifications.saved": "Preferences updated",
  "settings.options.currencies.EUR": "Euro (EUR)",
  "settings.options.currencies.GBP": "Pound Sterling (GBP)",
  "settings.options.currencies.USD": "US Dollar (USD)",
  "settings.options.locales.en": "English",
  "settings.options.locales.fr": "French",
  "settings.pageDescription": "Manage localization and accounting defaults for your workspace.",
  "settings.pageTitle": "Preferences",
  "settings.sections.currency.description": "Amounts are formatted using the selected currency.",
  "settings.sections.currency.preview": "Preview: {amount} on {date}",
  "settings.sections.currency.selectLabel": "Currency",
  "settings.sections.currency.title": "Accounting currency",
  "settings.sections.locale.description": "Choose the language used across the application.",
  "settings.sections.locale.selectLabel": "Language",
  "settings.sections.locale.title": "Interface language",
}
`;

exports[`i18n messages > matches the snapshot for French translations 1`] = `
{
  "actions.cancel": "Annuler",
  "actions.newEntry": "Nouvelle écriture",
  "actions.save": "Enregistrer",
  "app.footer.copyright": "© {year} Asso. Tous droits réservés.",
  "app.footer.links.legal": "Mentions légales",
  "app.footer.links.privacy": "Politique de confidentialité",
  "app.footer.links.support": "Support",
  "app.name": "Asso",
  "app.navigation.accounting": "Comptabilité",
  "app.navigation.dashboard": "Tableau de bord",
  "app.navigation.grants": "Subventions",
  "app.navigation.main": "Navigation principale",
  "app.navigation.memberPortal": "Portail adhérent",
  "app.navigation.members": "Membres",
  "app.navigation.projects": "Projets",
  "app.navigation.settings": "Paramètres",
  "app.navigation.sidebar": "Navigation latérale",
  "app.navigation.supervision": "Supervision",
  "app.navigation.title": "Navigation",
  "app.tagline": "Gestion & Comptabilité",
  "auth.logout": "Se déconnecter",
  "common.loading": "Chargement…",
  "common.modal.close": "Fermer",
  "layout.close": "Fermer",
  "layout.openNavigation": "Ouvrir la navigation",
  "settings.notifications.saved": "Préférences mises à jour",
  "settings.options.currencies.EUR": "Euro (EUR)",
  "settings.options.currencies.GBP": "Livre sterling (GBP)",
  "settings.options.currencies.USD": "Dollar américain (USD)",
  "settings.options.locales.en": "Anglais",
  "settings.options.locales.fr": "Français",
  "settings.pageDescription": "Configurez la langue et la devise utilisées dans l'application.",
  "settings.pageTitle": "Préférences",
  "settings.sections.currency.description": "Les montants sont formatés avec la devise sélectionnée.",
  "settings.sections.currency.preview": "Aperçu : {amount} au {date}",
  "settings.sections.currency.selectLabel": "Devise",
  "settings.sections.currency.title": "Devise comptable",
  "settings.sections.locale.description": "Choisissez la langue utilisée dans l'interface.",
  "settings.sections.locale.selectLabel": "Langue",
  "settings.sections.locale.title": "Langue de l'interface",
}
`;
